File: backend/src/main/java/com/backend/api/user/dto/response/UserSignupResponse.java
package com.backend.api.user.dto.response;

import com.backend.domain.ranking.entity.Ranking;
import com.backend.domain.ranking.entity.Tier;
import com.backend.domain.subscription.entity.SubscriptionType;
import com.backend.domain.user.entity.User;
import io.swagger.v3.oas.annotations.media.Schema;

public record UserSignupResponse(
        @Schema(description = "사용자 ID", example = "1")
        Long id,

        @Schema(description = "사용자 이메일", example = "user@example.com")
        String email,

        @Schema(description = "사용자 이름", example = "홍길동")
        String name,

        @Schema(description = "사용자 닉네임", example = "spring_dev")
        String nickname,

        @Schema(description = "사용자 구독 유형", example = "BASIC")
        SubscriptionType subscriptionType,

        @Schema(description = "사용자 티어", example = "UNRATED")
        Tier tier
) {
    public static UserSignupResponse from(User user, Ranking ranking) {
        return new UserSignupResponse(
                user.getId(),
                user.getEmail(),
                user.getName(),
                user.getNickname(),
                user.getSubscription().getSubscriptionType(),
                ranking.getTier()
        );
    }
}

File: backend/src/main/java/com/backend/api/user/dto/response/UserSignupResponse.kt
package com.backend.api.user.dto.response

import com.backend.domain.ranking.entity.Ranking
import com.backend.domain.ranking.entity.Tier
import com.backend.domain.subscription.entity.SubscriptionType
import com.backend.domain.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema

@JvmRecord
data class UserSignupResponse(
    @field:Schema(description = "사용자 ID", example = "1")
     val id: Long,

    @field:Schema(description = "사용자 이메일", example = "user@example.com")
    val email: String,

    @field:Schema(description = "사용자 이름", example = "홍길동")
    val name: String,

    @field:Schema(description = "사용자 닉네임", example = "spring_dev")
    val nickname: String,

    @field:Schema(description = "사용자 구독 유형", example = "BASIC")
    val subscriptionType: SubscriptionType,

    @field:Schema(description = "사용자 티어", example = "UNRATED")
    val tier: Tier

) {
    companion object {
        fun from(user: User, ranking: Ranking): UserSignupResponse {
            return UserSignupResponse(
                user.id,
                user.email,
                user.name,
                user.nickname,
                user.subscription!!.subscriptionType,
                ranking.tier
            )
        }
    }
}